openedx/features/student_certificates/helpers.py

```diff
@@ -113,17 +118,23 @@ def get_credential_certificates(user):
         certificate_url = credential.get('certificate_url')
         if not certificate_url:
             continue
-        program_name = [attribute.get('value') for attribute in credential.get('attributes', [])
-                        if attribute.get('name') == 'program_name']
-        if program_name:
-            completion_date = datetime.strptime(credential.get('created'), CREDENTIALS_DATE_FORMAT)
-            certificates.append({
-                'display_name': program_name[0],
-                'certificate_title': program_name[0],
-                'certificate_url': certificate_url,
-                'completion_date': completion_date.strftime(COMPLETION_DATE_FORMAT),
-                'is_program_cert': True,
-            })
+
+        program_uuid = credential['credential']['program_uuid']
+        program = cache.get(PROGRAM_CACHE_KEY_TPL.format(uuid=program_uuid))
+        if not program:
+            log.error('Program not found! Cache might be empty or don\'t contains program against uuid: %s',
+                      program_uuid)
```

Comment:
please use format instead of %s

Reply:
* `doesn\'t`

Reply:
Kindly specify a reason why `format` should be used here?

Reply:
@ahmed-faraz46 Done

Reply:
@muak50 I am not sure if we have set some convention or not, but at most of the places we are using format, and we should decide on something and document it in our code review document.

Reply:
@ahmed-faraz46 We are using both conventions. If we are using multiple place-holders than '.format' seems to be more appropriate with placed parameters. Here we are dealing with just one place-holder string, so both can be used here.

Reply:
@muak50 
%s is old style string formatting and .format is new style formating. And now in our new code we always prefer to use .format for formating string.

Reply:
https://realpython.com/python-string-formatting/

Reply:
@zeshanarifios I updated code based on your suggestion. Thanks